contracts/BundleExecutor.sol

```diff
@@ -41,35 +28,76 @@ contract FlashBotsMultiCall {
         _;
     }
 
-    constructor(address _executor) public payable {
+    constructor(ILendingPoolAddressesProvider _addressProvider) FlashLoanReceiverBase(_addressProvider) public payable {
         owner = msg.sender;
-        executor = _executor;
-        if (msg.value > 0) {
-            WETH.deposit{value: msg.value}();
-        }
+        WETH.approve(address(LENDING_POOL), uint(-1));
     }
 
-    receive() external payable {
+    /**
+        This function is called after your contract has received the flash loaned amount
+     */
+    function executeOperation(
+        address[] calldata assets,
+        uint256[] calldata amounts,
+        uint256[] calldata premiums,
+        address initiator,
+        bytes calldata params
+    )
+        external
+        override
+        returns (bool)
+    {
+        uint aaveDebt = amounts[0].add(premiums[0]);
```

Comment:
I guess since there's no assets held in the contract, you don't need to limit msg.sender to aave?